test/api/junction_detection_methods_test.cpp

```diff
@@ -86,9 +86,9 @@ TEST(junction_detection, single_method_only)
 
 TEST(junction_detection, method_pairs)
 {
-    for (uint8_t method_i : all_methods)
+    for (detection_methods method_i : all_methods)
     {
-        for (uint8_t method_j : all_methods)
+        for (detection_methods method_j : all_methods)
         {
```

Comment:
I'm not completely happy with the structure of these tests checking different method combinations. Do we really need the nested for-loops when the if-statements below check each combination explicitly anyways? I think the test could be made a little simpler without the for-loops but it's not a big deal and we can leave it like it is.

Reply:
Yes I understand what you mean, but I thought with this solution the tests will be a little grouped and created more automatically, which may avoid mistakes.
I would leave it like this, but of course you can always make a PR with better tests. :wink:

Reply:
Yes, I see your point and as the test does what it promises to do we can leave it as it is.